src/cuda/cuda-device.cpp

```diff
@@ -416,9 +418,24 @@ Result DeviceImpl::initialize(const DeviceDesc& desc)
     m_queue = new CommandQueueImpl(this, QueueType::Graphics);
     m_queue->setInternalReferenceCount(1);
 
-    SLANG_RETURN_ON_FAIL(m_clearEngine.initialize(m_debugCallback));
+    // Create 2 heaps. On CUDA both Upload and ReadBack just use host memory,
+    // so we only need one for DeviceLocal and one for Upload/ReadBack.
+    ComPtr<IHeap> heapPtr;
+    HeapDesc heapDesc = {};
+
+    heapDesc.memoryType = MemoryType::Upload;
+    heapDesc.label = "Device local heap";
+    SLANG_RETURN_ON_FAIL(createHeap(heapDesc, heapPtr.writeRef()));
+    m_hostMemHeap = checked_cast<HeapImpl*>(heapPtr.get());
```

Comment:
should we call this `m_uploadHeap` instead?

Reply:
Ah I see, you're sharing the heap for both upload and readback. Guess that's fine then. I still feel `MemHeap` is strange when the class is called just `Heap` now. I'd go for `m_deviceHeap` and `m_hostHeap` but no biggie.